aten/src/ATen/native/native_functions.yaml

```diff
@@ -3964,13 +3966,15 @@
 
 - func: set_.source_Tensor(Tensor(a!) self, Tensor source) -> Tensor(a!)
   variants: method
+  reset_grad_accumulator: True
   device_guard: False
   dispatch:
     CPU: set_tensor_
     CUDA: set_tensor_
 
 - func: set_(Tensor(a!) self) -> Tensor(a!)
   variants: method
+  reset_grad_accumulator: True
```

Comment:
It's a bit naughty (the way our code is currently setup) to control how VariableType codegen works from `native_functions.yaml`). I probably would have just opted for a blacklist in the relevant codegen script. Let's see how it goes.